sound/hda/intel-nhlt.c

```diff
@@ -310,6 +310,16 @@ intel_nhlt_get_endpoint_blob(struct device *dev, struct nhlt_acpi_table *nhlt,
 			cfg = nhlt_get_specific_cfg(dev, fmt, num_ch, rate, vbps, bps);
 			if (cfg)
 				return cfg;
+
+			/*
+			 * The DMIC bps=32 configuration is described in some
+			 * table with vbps=32 (the DMIC hardware is vbps=24)
+			 */
```

Comment:
@ujfalusi this patch feels a bit odd. May I suggest a different approach in the SOF driver by looking up both 32/32 and 24/32 when the bit-depth is 32 instead?

Reply:
@ranj063, I had that implementation first, but it felt out of place and hackish. The reason why I have moved it here is that it is the NHLT table itself which is incorrect and it is better to handle this in the NHLT code rather then in upper layer.

Reply:
no, the NHLT is not incorrect. It's really that vbs is irrelevant and should be ignored.

Reply:
OK, so rewording the comments and the commit message would suffice?
But then if the user asks for 32/32 then we need to look for 32/24 also, right? For DMIC, that is.
The NHLT core can do this:
if DMIC and bps == 32, set the vbps to 24 first, try to find the blob, if not found, look for it using vbps = 32.
Or pass  vbps = 0 to `nhlt_get_specific_cfg()` and if vbps is 0, do not try to match it, only look for bps match.

Reply:
> pass vbps = 0 to `nhlt_get_specific_cfg()` and if vbps is 0, do not try to match it, only look for bps match.

that would be my preference. vbps is a wildcard - ignored.

Reply:
me too!

Reply:
I'm ok with this as long as we can limit this to DMIC in the caller (i.e. pass 0 to use a wildcard).